### PR TITLE
adds ability to phase dsets together

### PR DESCRIPTION
**Pull request by nkern**
Adds a `rephase_to_dset()` method on a PSpecData instance. This allows us to form (more) coherent power spectra when forming cross-LST bin spectra. This interfaces with `hera_cal.lstbin.lst_rephase`. Addresses #44.

### CODE REVIEW

File: hera_pspec/pspecdata.py [hunk 1]
```diff
@@ -819,3 +823,77 @@ def pspec(self, bls, beam=None, input_data_weight='identity', norm='I',
                     pairs.append((key1, key2))
                     
         return np.array(pvs), pairs
+
+    def rephase_to_dset(self, dset_index, inplace=True):
+        """
+        Rephase data in dsets to the LST grid of dset[dset_index] using
+        hera_cal.lstbin.lst_rephase.
```
Comment (philbull) [inline hera_pspec/pspecdata.py]: Please add a few simple details about what's happening here, e.g. are the LSTs being rephased to the centre of the reference LST bins?

File: hera_pspec/pspecdata.py [hunk 2]
```diff
@@ -819,3 +823,77 @@ def pspec(self, bls, beam=None, input_data_weight='identity', norm='I',
...
+        Parameters
+        ----------
+        dset_index : int
+            index of dataset in self.dset to phase other datasets to.
```
Comment (philbull) [inline hera_pspec/pspecdata.py]: Maybe this could have a default of 0?

File: hera_pspec/pspecdata.py [hunk 3]
```diff
@@ -819,3 +823,77 @@ def pspec(self, bls, beam=None, input_data_weight='identity', norm='I',
...
+            # rephase
+            hc.lstbin.lst_rephase(data, bls, freqs, dlst, lat=lat)
+
+            # re-insert into dataset
+            for j, k in enumerate(data.keys()):
+                indices = dset.antpair2ind(*k[:2])
+                polind = pol_list.index(hc.io.polstr2num[k[-1]])
+                dset.data_array[indices, 0, :, polind] = data[k]
+
+            # set phasing to unknown
```
Comment (philbull) [inline hera_pspec/pspecdata.py]: Please explain what this is for.

**Comment by acliu** [comment]
Phil, a typo: "this is nice addition" --> "this is a nice addition" ;)

File: hera_pspec/pspecdata.py (hunk 2 above)
Comment (nkern) [reply hera_pspec/pspecdata.py]: Sure

**Comment by nkern** [comment]
actually, @acliu, people call me addition sometimes

**Pull request merged**